File: Csharp/diagonalNegativos.cs
using System;

namespace diagonalNegativos {
    internal class Program {
        static void Main(string[] args) {

            int N, i, j;

            Console.Write("Qual a ordem da matriz? ");
            N = int.Parse(Console.ReadLine());

            int[,] mat = new int[N, N];

            // Entrada de dados
            for (i = 0; i < N; i++) {
                for (j = 0; j < N; j++) {
                    Console.Write("Elemento [" + i + "," + j + "]: ");
                    mat[i, j] = int.Parse(Console.ReadLine());
                }
            }

            Console.WriteLine();
            Console.WriteLine("DIAGONAL PRINCIPAL:");
            // Diagonal Principal
            for (i = 0; i < N; i++) {
                Console.Write(mat[i, i] + " ");
            }

            // Quantidade de Negativos

            int count = 0;
            for (i = 0; i < N; i++) {
                for (j = 0; j < N; j++) {
                    if (mat[i, j] < 0) {
                        count++;
                    }
                }
            }

            Console.WriteLine();
            Console.WriteLine("QUANTIDADE DE NEGATIVOS = " + count);
        }
    }
}

File: Cpp/diagonal_negativos.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){

    int N, i, j;

    cout << "Qual a ordem da matriz? ";
    cin >> N;

    int mat[N][N];

    for(i=0; i<N; i++){
        for(j=0; j<N; j++){
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> mat[i][j];
        }
    }

    cout << endl << "DIAGONAL PRINCIPAL: " << endl;

    for(i=0; i<N; i++){
        cout << mat[i][i] << " ";
    }

    int cont = 0;
    for(i=0; i<N; i++){
        for(j=0; j<N; j++){
            if(mat[i][j]<0){
                cont++;
            }
        }
    }

    cout << endl << "QUANTIDADE DE NEGATIVOS = " << cont << endl;

return 0;
}
